src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -885,6 +889,176 @@ MavlinkReceiver::handle_message_set_position_target_local_ned(mavlink_message_t
 	}
 }
 
+void
+MavlinkReceiver::handle_message_set_position_target_global_int(mavlink_message_t *msg)
+{
+	mavlink_set_position_target_global_int_t set_position_target_global_int;
+	mavlink_msg_set_position_target_global_int_decode(msg, &set_position_target_global_int);
+
+	const bool values_finite =
+		PX4_ISFINITE(set_position_target_global_int.alt) &&
+		PX4_ISFINITE(set_position_target_global_int.vx) &&
+		PX4_ISFINITE(set_position_target_global_int.vy) &&
+		PX4_ISFINITE(set_position_target_global_int.vz) &&
+		PX4_ISFINITE(set_position_target_global_int.afx) &&
+		PX4_ISFINITE(set_position_target_global_int.afy) &&
+		PX4_ISFINITE(set_position_target_global_int.afz) &&
+		PX4_ISFINITE(set_position_target_global_int.yaw);
+
+	/* Only accept messages which are intended for this system */
+	if ((mavlink_system.sysid == set_position_target_global_int.target_system ||
+	     set_position_target_global_int.target_system == 0) &&
+	    (mavlink_system.compid == set_position_target_global_int.target_component ||
+	     set_position_target_global_int.target_component == 0) &&
+	    values_finite) {
```

Comment:
I would prefer less indentation levels. So instead of:
```
if (condition1) {
    if (condition2) {
       do_stuff();
    }
}
```

you could often use:
```
if (!condition1) {
    return;
}
if (!condition2) {
    return;
}
do_stuff();
```

Reply:
I was following the logic from `handle_message_set_position_target_local_ned(mavlink_message_t *msg)` Wouldn't it make sense to refactor both in that case?

Reply:
If this is the case, I would prefer modifying the logic together in a separate PR

Reply:
Fine if it's done at a later point.

Reply:
Thanks